core/payments/payments.go

```diff
@@ -218,9 +218,11 @@ func (s *service) GetSubscriptionStatus(ctx context.Context, req *pb.RpcMembersh
 
 	// if NoCache flag -> skip returning from cache
 	if !req.NoCache && (err == nil) && (cachedStatus != nil) && (cachedStatus.Data != nil) {
-		// 2. If found in cache -> return it
-		log.Debug("returning subscription status from cache", zap.Error(err), zap.Any("cachedStatus", cachedStatus))
-		return cachedStatus, nil
+		if (cachedStatus.Error == nil) || (cachedStatus.Error.Code == pb.RpcMembershipGetStatusResponseError_NULL) {
```

Comment:
Maybe we should not call **s.cache.CacheGet** on every request?
Imho we can call it in two cases: when NoCache is not set and when request to PP node is failed. Then code can be simplified a lil bit:

```
	canReturnCachedStatus := func(s *pb.RpcMembershipGetStatusResponse) bool {
		return s != nil && s.Data != nil && (s.Error == nil || s.Error.Code == pb.RpcMembershipGetStatusResponseError_NULL)
	}
	
	// 1 - check in cache
	if !req.NoCache {
		// if cache is disabled -> will return objects and ErrCacheDisabled
		// if cache is expired -> will return objects and ErrCacheExpired
		cachedStatus, _, err := s.cache.CacheGet()
		if err == nil && canReturnCachedStatus(cachedStatus) {
			log.Debug("returning subscription status from cache", zap.Error(err), zap.Any("cachedStatus", cachedStatus))
			return cachedStatus, nil
		}
	}

	...

	status, err := s.ppclient.GetSubscriptionStatus(ctx, &reqSigned)
	if err != nil {
		// 4a. try reading from cache again
		cachedStatus, _, err := s.cache.CacheGet()
		if err == nil && canReturnCachedStatus(cachedStatus) {
			log.Debug("returning subscription status from cache", zap.Error(err), zap.Any("cachedStatus", cachedStatus))
			return cachedStatus, nil
		}

		...
	}


```